code/parachain/runtime/primitives/src/currency.rs

```diff
@@ -89,22 +89,12 @@ macro_rules! list_assets {
 			}
 		}
 
-		const RELAY_NATIVE_CURRENCIES: [CurrencyId; 1] = [CurrencyId::KSM];
-
-		fn known_location(id: CurrencyId) -> Option<XcmAssetLocation> {
-			if Self::RELAY_NATIVE_CURRENCIES.contains(&id) {
-				return Some(XcmAssetLocation::RELAY_NATIVE)
-			}
-			// Note: add other known locations
-			None
-		}
-
 		pub fn list_assets() -> Vec<Asset<XcmAssetLocation>> {
 			[
 				$(Asset {
 					id: CurrencyId::$NAME.0 as u128,
 					name: Some(stringify!($NAME).as_bytes().to_vec()),
-					decimals: 12_u8.into(),
+					decimals: Self::decimals(),
```

Comment:
not sure if we can break UI now, but better assumption could be if decimals are made Option

Reply:
but that just wish, very low prio and not blocker.

Reply:
As far as I understand, we are not planning to include assets that don't have decimals. At least not for now.